duplicacy-autobackup.sh

```diff
@@ -28,26 +28,38 @@ do_backup() {
     echo "Running pre-backup script"
     sh $PRE_BACKUP_SCRIPT
     status=$?
-  fi
-  if [[ $status != 0 ]]; then
-    echo "Pre-backup script exited with status code $status. Not performing backup." >&2
-    return
+    if [[ $status != 0 ]]; then
+      echo "Pre-backup script exited with status code $status. Not performing backup." >&2
+      exit $status
+    fi
   fi
 
   duplicacy backup $DUPLICACY_BACKUP_OPTIONS
+  status=$?
+  if [[ $status != 0 ]]; then
+    echo "Duplicacy backup failed." >&2
+    exit $status
+  fi
 
   if [[ -f $POST_BACKUP_SCRIPT ]]; then
     echo "Running post-backup script"
     sh $POST_BACKUP_SCRIPT
     status=$?
     echo "Post-backup script exited with status $status"
+    exit $status
```

Comment:
I would suggest exiting here only if $status != 0. If leaves room for adding more code to the script